tokio-timer/src/debounce.rs

```diff
@@ -0,0 +1,408 @@
+//! Debounce streams on the leading or trailing edge or both edges for a certain
```

Comment:
This implementation should not be in `tokio-timer`. The `tokio-timer crate should be reserved for foundational types that either cannot be implemented outside of the crate **or** have trivial / obvious implementations.

I do not believe debounce satisfies this, so it should live in tokio proper.

Reply:
Would you be fine with the tests remaining in tokio_timer? Currently they require some code from `tokio_timer::tests::support` to run.